From f2fb2c14ea4a5181468af99c2dd74166c9a26e49 Mon Sep 17 00:00:00 2001
From: kpandu549 <keerthana_pandurangan@comcast.com>
Date: Fri, 10 Dec 2021 08:56:43 +0000
Subject: [PATCH] REFPLTB-1488 : RPI-B not getting erouter IP with topic
 cmf-update-RDKB-35566-2107_sprint-stable2-20211209

Reason for change: Unable to get the WAN interface ip due to CcspEthAgent crash.
Test Procedure: erouter0 interface is able to get IP address
Risks: Low

Signed-off-by: kpandu549 <keerthana_pandurangan@comcast.com>
Change-Id: I84954ec90dcfeae6aef3bb3dbe964bfd7c48f98d
---
 source/TR-181/middle_layer_src/cosa_ethernet_internal.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/source/TR-181/middle_layer_src/cosa_ethernet_internal.c b/source/TR-181/middle_layer_src/cosa_ethernet_internal.c
index a411076..5165e16 100644
--- a/source/TR-181/middle_layer_src/cosa_ethernet_internal.c
+++ b/source/TR-181/middle_layer_src/cosa_ethernet_internal.c
@@ -439,7 +439,9 @@ CosaEthernetInitialize
     CosaDmlEthWanGetCfg(&pMyObject->EthWanCfg);
 
 #if defined(FEATURE_RDKB_WAN_MANAGER)
-
+#if defined(_PLATFORM_RASPBERRYPI_) || defined(_PLATFORM_TURRIS_)
+    CosaDmlEthInit(NULL, (PANSC_HANDLE)pMyObject);
+#endif    
     //Register callbacks with ethsw-hal for link events.
     appCallBack obj;
     memset (&obj, 0, sizeof(appCallBack));
@@ -447,8 +449,10 @@ CosaEthernetInitialize
     obj.pGWP_act_EthWanLinkUP   = EthWanLinkUp_callback;
     GWP_RegisterEthWan_Callback ( &obj );
 
+#if !defined(_PLATFORM_RASPBERRYPI_) && !defined(_PLATFORM_TURRIS_)
     //Initialise global data and initalise hal
     CosaDmlEthInit(NULL, (PANSC_HANDLE)pMyObject);
+#endif    
 
     #if defined(_COSA_BCM_ARM_)  && defined(_XB6_PRODUCT_REQ_)
             #if !defined(_PLATFORM_RASPBERRYPI_) && !defined(_PLATFORM_TURRIS_)
-- 
2.28.0

